tools/render-test/render.cpp

```diff
@@ -328,4 +365,62 @@ void TextureResource::Desc::init3D(Format formatIn, int widthIn, int heightIn, i
     this->cpuAccessFlags = 0;
 }
 
+/* !!!!!!!!!!!!!!!!!!!!!!!!! RennderUtil !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! */
+
+ProjectionStyle RendererUtil::getProjectionStyle(RendererType type)
+{
+    switch (type)
+    {
+        case RendererType::DirectX11:
+        case RendererType::DirectX12:
+        {
+            return ProjectionStyle::DirectX;
+        }
+        case RendererType::OpenGl:              return ProjectionStyle::OpenGl;
+        case RendererType::Vulkan:              return ProjectionStyle::Vulkan;
+        case RendererType::Unknown:             return ProjectionStyle::Unknown;
+        default:
+        {
+            assert(!"Unhandled type");
+            return ProjectionStyle::Unknown;
+        }
+    }
+}
+
+/* static */void RendererUtil::getIdentityProjection(ProjectionStyle style, float projMatrix[16])
+{
+    switch (style)
+    {
+        case ProjectionStyle::DirectX:
+        case ProjectionStyle::OpenGl:
+        {
+            static const float kIdentity[] =
+            { 
+                1, 0, 0, 0,
+                0, 1, 0, 0,
+                0, 0, 1, 0,
+                0, 0, 0, 1
+            };
+            ::memcpy(projMatrix, kIdentity, sizeof(kIdentity));
+            break;
+        }
+        case ProjectionStyle::Vulkan:
```

Comment:
First point - understood. It was the minimum to get things working. As it standings culling is disabled across all Renderers so that's not a problem, right now. 

Second point - agreed. But as you say that is another can of worms, and Renderer has no support to rendering to a Texture so for now it seems acceptable. 

```Also: why make this a utility function hoisted out of the Renderer interface (that nonetheless needs to have a case for each and every Renderer implementation), instead of just a virtual method in Renderer?```

Because stuff that goes on inside the renderer is obscure/hard code to read. That if it is going to produce a projection matrix ... well there are lots of different ways to do that for different reasons. So am I going to create a function/s on a renderer to create every kind of projection? So here the 'magic' is exposed. I don't need to pass in all the parameters to the Renderer to produce a projection. I don't even need a Renderer.